server/webapp/WEB-INF/rails.new/app/presenters/api_v1/config/artifact_representer.rb

```diff
@@ -39,17 +39,21 @@ def type
       end
 
       def errors
-        mapped_errors = artifact.errors
-
-        if src_errors = mapped_errors.delete('src')
-          mapped_errors['source'] = src_errors
+        mapped_errors = {}
+        artifact.errors.each do |key, value|
+          mapped_errors[matching_error_key(key)] = value
         end
+        mapped_errors
+      end
 
-        if dest_errors = mapped_errors.delete('dest')
-          mapped_errors['destination'] = dest_errors
-        end
+      private
+      def error_keys
+        {"src" => "source", "dest" => "destination"}
+      end
 
-        mapped_errors
+      def matching_error_key key
+        return error_keys[key] if error_keys[key]
+        key
```

Comment:
On second thought — this DSL probably reads better —

```ruby
class FooRepresenter < BaseRepresenter
    has_errors('src' => 'source', 'dest' => 'destination')
end
```

and let `has_errors` meta-program the necessary methods. Ping me or @maheshp if you'd like to understand this better.